sqlglot/dialects/snowflake.py

```diff
@@ -114,6 +114,13 @@ def _build_bitor(args: t.List) -> exp.BitwiseOr | exp.Anonymous:
     return binary_from_function(exp.BitwiseOr)(args)
 
 
+def _build_bitxor(args: t.List) -> exp.BitwiseXor | exp.Anonymous:
+    if len(args) == 3:
+        return exp.Anonymous(this="BITXOR", expressions=args)
+
+    return binary_from_function(exp.BitwiseXor)(args)
+
+
```

Comment:
This is largely repeating the `BITOR` implementation, we could merge these two under a single builder function that takes in (1) the anonymous name and (2) the `expr_type` of the node to-be-built, e.g roughly:

```Python3
def _build_bitwise(expr_type: t.Type[E], name: str) -> t.Callable[t.List, E | exp.Anonymous]:
    def _builder(args: t.List, dialect: Dialect) -> E:
        if len(args) == 3:
            return exp.Anonymous(this=name, expressions=args)

        return binary_from_function(expr_type)(args)

    return _builder
```



Check out other builders like `dialect.py::build_regexp_extract()` for similar ideas.

Reply:
@VaggelisD Made the changes. Thanks for this good suggestion. I should have merged these functions and created a reusable function in the previous commit itself. Better late than never 😄

Reply:
No worries, these are patterns that usually occur only after we have accumulated repetitive sections. Your initial solution was fine too 👍